Tasks/Xcode/xcodeutils.ts

```diff
@@ -187,6 +187,25 @@ export function pathExistsAsFile(path: string) {
     }
 }
 
+export function pipeOutputToFile(firstTool: ToolRunner, secondTool: ToolRunner, logPrefix: string) : string {
```

Comment:
I think this function's name is a little confusing, since we are piping both to `secondTool` and some file created from `logPrefix`.  I think this is because we are trying to do two things at once: create the unique log file name and pipe output.  If something is too hard to name, I find it's usually a smell that it should be more modular.

So I think this is the function we want to export:

```TypeScript
/** Find an unused file name of the form $(Agent.TempDirectory) + logPrefix + <number> + .log. */
export function createUniqueLogFileName(logPrefix: string): string {
    let filePath: string = tl.resolve(tl.getVariable('Agent.TempDirectory'), logPrefix + '.log');
    let index = 1;
    while (tl.exist(filePath)) {
        filePath = tl.resolve(tl.getVariable('Agent.TempDirectory'), logPrefix + index.toString() + '.log');
        index++;
    }
}
```

Then the call site looks like:
```TypeScript
const logFile: string = utils.createUniqueLogFileName('xcodebuild_archive');
xcodeArchive.pipeExecOutputToTool(xcPrettyTool, logFile);
utils.setTaskState('XCODEBUILD_ARCHIVE_LOG', logFile);
```

It's an extra LOC but I find it easier to understand the logic.